src/ncp/ncp_buffer.hpp

```diff
@@ -39,11 +39,33 @@
 
 namespace ot {
 
+
+/**
+ * This class implements a buffer/queue for storing NCP frames.
+ *
+ * A frame can consist of a sequence of data bytes and/or the content of an `otMessage` or a combination of the two.
+ * NcoFrameBuffer implements priority FIFO logic for storing and reading frames. Two priority levels of high and low
+ * are supported. Within same priority level first-in-first-out order is preserved. High priority frames are read
+ * ahead of the low priority ones.
+ *
+ */
 class NcpFrameBuffer
 {
 public:
+
+    /**
+     * Defines the priority of frame. High priority frames are read before low priority frames. Within same priority
+     * level FIFO order is preserved.
+     *
+     */
+    enum Priority
+    {
+        kPriorityLow  = 0,              //< Indicates low/normal priority for a frame.
+        kPriorityHigh = 1,              //< Indicates high priority for a frame.
+    };
```

Comment:
We are using the `enum Priority` as a type itself (e.g., as input argument to `InFrameBegin(Priority aPriority)`, so my idea was to limit the definitions within this enum to only valid priority types (to keep the type/value safety check by compiler when API are used).